Obsidian/Util/PacketHandler.cs

```diff
@@ -67,528 +37,69 @@ public static async Task<Packet> ReadCompressedPacketAsync(MinecraftStream strea
-                                Animation = EAnimation.SwingMainArm
-                            }, player);
-                            break;
-                        case Hand.OffHand:
-                            await server.BroadcastPacketAsync(new EntityAnimation
-                            {
-                                EntityId = client.id,
-                                Animation = EAnimation.SwingOffhand
-                            }, player);
-                            break;
-                        default:
-                            break;
-                    }
-                    break;
-
-                case 0x2D://Spectate
-                    break;
-                case 0x2E:// Player Block Placement
-                    {
-                        var pbp = PacketSerializer.FastDeserialize<PlayerBlockPlacement>(packet.data);
-
-                        var currentItem = player.GetHeldItem();
-
-                        var block = Registry.GetBlock(currentItem.Type);
-
-                        var location = pbp.Location;
-
-                        var interactedBlock = server.World.GetBlock(location);
-
-                        if (interactedBlock.CanInteract() && !player.Sneaking)
-                        {
-                            var arg = await server.Events.InvokeBlockInteractAsync(new BlockInteractEventArgs(player, block, pbp.Location));
-
-                            if (arg.Cancel)
-                                return;
-
-                            //TODO open chests/Crafting inventory ^ ^
-
-                            Logger.LogDebug($"Block Interact: {interactedBlock} - {location}");
-
-                            return;
-                        }
-
-                        if (player.Gamemode != Gamemode.Creative)
-                            player.Inventory.RemoveItem(player.CurrentSlot);
-
-                        switch (pbp.Face) // TODO fix this for logs
-                        {
-                            case BlockFace.Bottom:
-                                location.Y -= 1;
-                                break;
-
-                            case BlockFace.Top:
-                                location.Y += 1;
-                                break;
-
-                            case BlockFace.North:
-                                location.Z -= 1;
-                                break;
-
-                            case BlockFace.South:
-                                location.Z += 1;
-                                break;
-
-                            case BlockFace.West:
-                                location.X -= 1;
-                                break;
-
-                            case BlockFace.East:
-                                location.X += 1;
-                                break;
-
-                            default:
-                                break;
-                        }
+        public static void RegisterHandlers()
```

Comment:
an idea: we could make an attribute for packets with an id and define default behavior for IPacket.Id to return the attribute's value. But keep it like this for now, I'll experiment with it once it hits stable.